### PR TITLE
Add a note about potential type promotion to interp_multidim

### LINKED ISSUE #488: `interp_multidim` can change data precision
When using `interp_multidim` the interpolated data precision can change based on the coordinate variable precision.

Specifically, we this came up in an ADF issue (https://github.com/NCAR/ADF/issues/192) where it was noted that data variables were getting promoted to double from single. 

It appears that this occurs whenever either the source or destination coordinate variables are double.

**Example that demonstrates the behavior**
```python
import numpy as np
import xarray as xr
import geocat.comp as gc

rng = np.random.default_rng()
fake_data = rng.random((90, 180), dtype=np.dtype('float32'))  
fake_data = xr.DataArray(fake_data, coords={"lat":np.linspace(-89, 89, 90), "lon":np.linspace(0,360-2,180)})
tgt_grid = xr.Dataset(coords={"lat":np.linspace(-89, 89, 45), "lon":np.linspace(0,360-4,90)})
regridded_data = gc.interp_multidim(fake_data, tgt_grid.lat.values, tgt_grid.lon.values, cyclic=True)
print(f"With float64 coordinates (all): original dtype: {fake_data.dtype}, output dtype: {regridded_data.dtype}")

fake_data = fake_data.assign_coords({"lat":np.linspace(-89, 89, 90).astype('float32'), "lon":np.linspace(0,360-2,180).astype('float32')})
tgt_grid = xr.Dataset(coords={"lat":np.linspace(-89, 89, 45), "lon":np.linspace(0,360-4,90)})
regridded_data = gc.interp_multidim(fake_data, tgt_grid.lat.values, tgt_grid.lon.values, cyclic=True)
print(f"Input coords are {fake_data.lat.dtype}, output coords are {tgt_grid.lat.dtype}: original dtype: {fake_data.dtype}, output dtype: {regridded_data.dtype}")

fake_data = fake_data.assign_coords({"lat":np.linspace(-89, 89, 90).astype('float32'), "lon":np.linspace(0,360-2,180).astype('float32')})
tgt_grid = xr.Dataset(coords={"lat":np.linspace(-89, 89, 45).astype('float32'), "lon":np.linspace(0,360-4,90).astype('float32')})
regridded_data = gc.interp_multidim(fake_data, tgt_grid.lat.values, tgt_grid.lon.values, cyclic=True)
print(f"Input coords are {fake_data.lat.dtype}, output coords are {tgt_grid.lat.dtype}: original dtype: {fake_data.dtype}, output dtype: {regridded_data.dtype}")

fake_data = fake_data.assign_coords({"lat":np.linspace(-89, 89, 90), "lon":np.linspace(0,360-2,180)})
tgt_grid = xr.Dataset(coords={"lat":np.linspace(-89, 89, 45).astype('float32'), "lon":np.linspace(0,360-4,90).astype('float32')})
regridded_data = gc.interp_multidim(fake_data, tgt_grid.lat.values, tgt_grid.lon.values, cyclic=True)
print(f"Input coords are {fake_data.lat.dtype}, output coords are {tgt_grid.lat.dtype}: original dtype: {fake_data.dtype}, output dtype: {regridded_data.dtype}")
```
**Result**
```
With float64 coordinates (all): original dtype: float32, output dtype: float64
Input coords are float32, output coords are float64: original dtype: float32, output dtype: float64
Input coords are float32, output coords are float32: original dtype: float32, output dtype: float32
Input coords are float64, output coords are float32: original dtype: float32, output dtype: float64
```

**Expected behavior**
The precision of the data should be preserved.

**Environment**
I don't think versions matter much for this, but I'm using:
- numpy 1.24.4
- xarray 2023.2.0
- geocat-comp 2023.10.0

I think this problem originates from the xarray methods being used, so maybe we need to ask them about it, too (?).

### PR DESCRIPTION
**Pull request by kafitzgerald**
## PR Summary
Add a note about potential type promotion to interp_multidim docs and fix a few typos.

Closes #488

## PR Checklist

**General**
- [x] Make an issue if one doesn't already exist
- [x] Link the issue this PR resolves by adding `closes #XXX` to the PR description where XXX is the number of the issue.
- [x] Add a brief summary of changes to `docs/release-notes.rst` in a relevant section for the next unreleased release. Possible sections include: Documentation, New Features, Bug Fixes, Internal Changes, Breaking Changes/Deprecated
- [x] Add appropriate labels to this PR
- [x] Make your changes in a forked repository rather than directly in this repo
- [x] Open this PR as a draft if it is not ready for review
- [x] Convert this PR from a draft to a full PR before requesting reviewers
- [x] Passes `precommit`. To set up on your local, run `pre-commit install` from the top level of the repository. To manually run pre-commits, use `pre-commit run --all-files` and re-add any changed files before committing again and pushing.
- [x] If needed, squash and merge PR commits into a single commit to clean up commit history

### CODE REVIEW

File: geocat/comp/interpolation.py [hunk 1]
```diff
@@ -662,16 +663,16 @@ def interp_multidim(
...
     -------
     data_out : ndarray, :class:`xarray.DataArray`
-       Returns same data type as input ``data_in``. Shape will be the same as
-       input array except
-       for last two dimensions which will be equal to thecoordinates given in
-       ``data_out``
+       Returns the same type of object as input ``data_in``. However, the type of
+       the data in the array may be promoted to that of the coordinates. Shape
+       will be the same as input array except for last two dimensions which will
+       be equal to the coordinates given in ``data_out``.
```
Comment (anissa111) [inline geocat/comp/interpolation.py:675]: Should we add this as a more prominent warning or note? I'm worried that it might get overlooked otherwise.

Comment (kafitzgerald) [reply geocat/comp/interpolation.py:675]: Probably.  

I'll link it to the xarray docs as well for more detailed info.

Comment (kafitzgerald) [reply geocat/comp/interpolation.py:677]: I went ahead and added a warning near the top of the description.  There's mention of this in the returns as well, but let me know if you think the warning would be better in that section.  

Unfortunately there's not really discussion of this in the xarray docs (or really even in scipy for that matter) and thankfully we've already linked them so I skipped over that.  We could add more description here, but I think this is probably sufficient and less likely to fall out of date than if I went into more detail.

Comment (anissa111) [reply geocat/comp/interpolation.py:677]: That looks good to me.

**Pull request merged**